move/operators/sources/operators/operators.move

```diff
@@ -0,0 +1,212 @@
+module operators::operators {
+    use sui::bag::{Self, Bag};
+
+    // -----
+    // Types
+    // -----
+
+    /// The `OwnerCap` capability representing the owner of the contract.
+    public struct OwnerCap has key, store {
+        id: UID,
+    }
+
+    /// The `OperatorCap` capability representing an approved operator.
+    public struct OperatorCap has key, store {
+        id: UID,
+    }
+
+    /// The main `Operators` struct storing the capabilities and operator IDs.
+    public struct Operators has key {
+        id: UID,
+        operator_ids: vector<ID>,
+        caps: Bag,
+    }
+
+    // -----
+    // Setup
+    // -----
+
+    /// Initializes the contract and transfers the `OwnerCap` to the sender.
+    fun init(ctx: &mut TxContext) {
+        transfer::share_object(Operators {
+            id: object::new(ctx),
+            operator_ids: vector::empty(),
+            caps: bag::new(ctx),
+        });
+
+        transfer::public_transfer(OwnerCap {
+            id: object::new(ctx),
+        }, tx_context::sender(ctx));
+    }
+
+    // ----------------
+    // Public Functions
+    // ----------------
+
+    /// Adds a new operator by issuing an `OperatorCap` and storing its ID.
+    public fun add_operator(self: &mut Operators, _: &OwnerCap, new_operator: address, ctx: &mut TxContext) {
+        let operator_cap = OperatorCap {
+            id: object::new(ctx),
+        };
+        let operator_id = object::id(&operator_cap);
+        transfer::public_transfer(operator_cap, new_operator);
+        self.operator_ids.push_back(operator_id);
+    }
+
+    /// Removes an operator by ID, revoking their `OperatorCap`.
+    public fun remove_operator(self: &mut Operators, _: &OwnerCap, operator_id: ID) {
+        let (found, index) = self.operator_ids.index_of(&operator_id);
+        assert!(found, 0);
+        self.operator_ids.remove(index);
+    }
+
+    /// Stores a capability in the `Operators` struct.
+    public fun store_cap<T: key + store>(self: &mut Operators, _: &OwnerCap, cap: T) {
+        let cap_id = object::id(&cap);
+        self.caps.add(cap_id, cap);
+    }
+
+    /// Allows an approved operator to borrow a capability by its ID.
+    public fun borrow_cap<T: key + store>(self: &Operators, operator_cap: &OperatorCap, cap_id: ID): &T {
+        let operator_id = object::id(operator_cap);
+
+        assert!(self.operator_ids.contains(&operator_id), 0);
+        assert!(self.caps.contains(cap_id), 1);
+
+        &self.caps[cap_id]
+    }
+
+    /// Removes a capability from the `Operators` struct.
+    public fun transfer_cap<T: key + store>(self: &mut Operators, _: &OwnerCap, cap_id: ID, new_holder: address) {
+        transfer::public_transfer(self.caps.remove<ID, T>(cap_id), new_holder);
```

Comment:
I suggest that this function just returns the cap instead and the owner can do as they please wtih it. Would probably rename this to get_cap or remove_cap as well.